### PR TITLE
Add uncontrolled mode for Tree

### PR DESCRIPTION
**Pull request by tdraier**
## Description

Related to https://github.com/dust-tt/dust/pull/5632 , dedicated PR for sparkle
Makes an uncontrolled version of collapse/expand to avoid managing state in every usage

## Risk

Old behaviour is still supported

## Deploy Plan

Build and deploy sparkle

### CODE REVIEW

File: sparkle/src/components/Tree.tsx [hunk 1]
```diff
@@ -61,9 +62,25 @@ Tree.Item = function ({
...
+  const [collapsedState, setCollapsedState] = useState(defaultCollapsed);
+
+  const controlled = typeof collapsed !== "undefined";
+
+  const effectiveCollapsed = controlled ? collapsed : collapsedState;
+  const effectiveOnChevronClick = controlled
+    ? onChevronClick
+    : () => setCollapsedState(!collapsedState);
+
+  const childrenToRender = !effectiveCollapsed
```
[the remark this answers is not in the document]
Comment (tdraier) [reply sparkle/src/components/Tree.tsx:78]: IIFE ? :-)

File: sparkle/src/components/Tree.tsx [hunk 2]
```diff
@@ -61,9 +62,25 @@ Tree.Item = function ({
...
   onChevronClick,
   collapsed,
-  children,
+  defaultCollapsed,
   actions,
+  children,
 }: TreeItemProps) {
+  const [collapsedState, setCollapsedState] = useState(defaultCollapsed);
+
+  const controlled = typeof collapsed !== "undefined";
```
[the remark this answers is not in the document]
Comment (tdraier) [reply sparkle/src/components/Tree.tsx:71]: what kind of check ? if it's false, it means it's not collapsed - only undefined means it's not controlled

File: sparkle/src/components/Tree.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (flvndvd) [reply sparkle/src/components/Tree.tsx]: 😱

File: sparkle/src/components/Tree.tsx [hunk 3]
```diff
@@ -39,19 +39,29 @@ const visualTable = {
...
+}
+
+export interface TreeItemPropsWithChildren extends TreeItemProps {
+  renderTreeItems?: undefined;
   children?: React.ReactNode;
 }
 
+export interface TreeItemPropsWithRender extends TreeItemProps {
+  renderTreeItems: () => React.ReactNode;
+  children?: undefined;
```
Comment (flvndvd) [inline sparkle/src/components/Tree.tsx:62]: `nit`
```suggestion
  children?: never;
```

File: sparkle/src/components/Tree.tsx [hunk 4]
```diff
@@ -61,9 +71,32 @@ Tree.Item = function ({
...
   collapsed,
-  children,
+  defaultCollapsed,
   actions,
-}: TreeItemProps) {
+  renderTreeItems,
+  children,
+}: TreeItemPropsWithChildren | TreeItemPropsWithRender) {
+  const [collapsedState, setCollapsedState] = useState<boolean>(
+    defaultCollapsed || true
```
Comment (flvndvd) [inline sparkle/src/components/Tree.tsx:80]:
```suggestion
    defaultCollapsed ?? true
```

**Review by flvndvd — APPROVED** [review]
LGTM.

Don't forget to run `npm version patch` to bump the version before merging.

**Pull request merged**